packages/react-vapor/src/components/table-hoc/TableRowConnected.tsx

```diff
@@ -78,9 +78,11 @@ const mapDispatchToProps = (dispatch: IDispatch, ownProps: ITableRowOwnProps): I
                 dispatch(TableHOCRowActions.toggleCollapsible(ownProps.id, true));
             }
         },
-        onUnmount: () => {
+        onUnmount: (isSelected?: boolean) => {
             dispatch(TableHOCRowActions.remove(ownProps.id));
-            dispatch(TableHOCRowActions.deselectAll(ownProps.tableId));
+            if (isSelected) {
+                dispatch(addActionsToActionBar(ownProps.tableId, []));

```

Comment:
```suggestion
                dispatch(TableHOCRowActions.deselectAll(ownProps.tableId));
```

Reply:
I've exactly used TableHOCRowActionsType.deselectAll in my previous code. But it caused some bugs, like in the description. https://coveord.atlassian.net/secure/RapidBoard.jspa?rapidView=569&modal=detail&selectedIssue=ADUI-4695 Then Germain discussed with me, maybe we should try dispatching deselect a single one row instead of deselectAll in the table. But the problem is, after removing a row while unmount, that doesn't make sense of deselect it, because it's already gone, the effect of button is simply for the actions. So I chose to dispatch(addActionsToActionBar(ownProps.tableId, [])); for empty actions.

Reply:
oh ok yeah makes sense